.pre-commit-config.yaml

```diff
@@ -0,0 +1,39 @@
+---
+  - repo: https://github.com/psf/black
+    rev: 19.3b0
+    hooks:
+      - id: black
+        args: [--skip-string-normalization]
+        language_version: python3.6
+
+  - repo: https://github.com/asottile/seed-isort-config
+    rev: v1.9.2
+    hooks:
+      - id: seed-isort-config
+
+  - repo: https://github.com/pre-commit/mirrors-isort
+    rev: v4.3.21
+    hooks:
+      - id: isort
+        language_version: python3.6
+
+  - repo: https://github.com/detailyang/pre-commit-shell
+    rev: v1.0.6
+    hooks:
+      - id: shell-lint
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.2.3
+    hooks:
+      - id: flake8
+        args: [--ignore, 'E203,E501,W503']
```

Comment:
Couple things here.

1. Why ignore `E203`?
2. `E501` is pretty useful IMO, but fixing it is beyond the scope of this PR. Do you have any plans to make the code `E501` compliant in the future?
3. Ideally these lint suppressors would be inline with the code on the specific lines it makes sense to ignore them. If you really need them to be global then you should put them in one of the standard flake8 config files so other tools can pick them up: https://flake8.pycqa.org/en/latest/user/configuration.html#configuration-locations

Reply:
1. Removed E203 as you suggested.
2, 3. Yes, will definitely do both in the next refactoring. Made a ticket for refactoring. https://encodedcc.atlassian.net/browse/PIP-1321